### PR TITLE
fix: check `long_windows` option  match window in sql

### LINKED ISSUE #1711: Long window Option verification
**Bug Description**
```sql
deploy w14 options(long_windows='w9090:2') SELECT id, c1, sum(c4) OVER w1 as w1_c4_sum FROM window01 WINDOW w1 AS (PARTITION BY window01.c1 ORDER BY window01.c7 ROWS BETWEEN 2 PRECEDING AND CURRENT ROW);
```

If the `long_windows` options don't match any window in the sql, the `deploy` still succeed. It'd be better to add a checking here and return `failed`  when this happens.

**NOTE**

Should be the checking before we do any real action (i.e., create procedure, create pre-aggr table)

### PR DESCRIPTION
**Pull request by nautaa**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
fix

* **What is the current behavior?** (You can also link to an open issue here)
close #1711 

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

File: src/sdk/sql_cluster_router.cc [hunk 1]
```diff
@@ -3131,6 +3131,13 @@ hybridse::sdk::Status SQLClusterRouter::HandleLongWindows(
         if (!extract_status.IsOK()) {
             return extract_status;
         }
+        std::set<std::string> distinct_long_window;
+        for (const auto& info : long_window_infos) {
+            distinct_long_window.insert(info.window_name_);
+        }
+        if (distinct_long_window.size() != long_window_map.size()) {
+            return {base::ReturnCode::kError, "long_windows option doesn't match window in sql"};
+        }
```
Comment (zhanghaohit) [inline src/sdk/sql_cluster_router.cc:3140]: not exactly like this.

we want to check any `window_name` declared in the `long_windows` options should exist in the window definitions.

**Review by zhanghaohit — CHANGES_REQUESTED** [review]

File: src/sdk/sql_cluster_test.cc [hunk 2]
```diff
@@ -465,52 +465,67 @@ TEST_F(SQLClusterTest, CreatePreAggrTable) {
...
+    {
+        std::string deploy_sql = "deploy test1 options(long_windows='w2:1000,w1:1d') select col1,"
+                                 " sum(col3) over w1 as w1_sum_col3 from " + base_table +
+                                 " WINDOW w1 AS (PARTITION BY col1 ORDER BY col2"
+                                 " ROWS_RANGE BETWEEN 20s PRECEDING AND CURRENT ROW);";
+        router->ExecuteSQL(base_db, "use " + base_db + ";", &status);
+        router->ExecuteSQL(base_db, deploy_sql, &status);
+        ASSERT_EQ(status.code, -1);
+        ASSERT_EQ(status.msg, "long_windows option doesn't match window in sql");
+    }
```
Comment (zhanghaohit) [inline src/sdk/sql_cluster_test.cc:528]: "deploy test1 options(long_windows='w_error:1d') select col1,"
                                 " sum(col3) over w1 as w1_sum_col3 from " + base_table +
                                 " WINDOW w1 AS (PARTITION BY col1 ORDER BY col2"
                                 " ROWS_RANGE BETWEEN 20s PRECEDING AND CURRENT ROW);";

maybe add another test if the `window_name` is wrongly put

File: src/sdk/sql_cluster_test.cc [hunk 3]
```diff
@@ -465,52 +465,78 @@ TEST_F(SQLClusterTest, CreatePreAggrTable) {
...
+        std::string deploy_sql = "deploy test1 options(long_windows='w_error:1d') select col1,"
+                                 " sum(col3) over w1 as w1_sum_col3 from " + base_table +
+                                 " WINDOW w1 AS (PARTITION BY col1 ORDER BY col2"
+                                 " ROWS_RANGE BETWEEN 20s PRECEDING AND CURRENT ROW);";
+        router->ExecuteSQL(base_db, "use " + base_db + ";", &status);
+        router->ExecuteSQL(base_db, deploy_sql, &status);
+        ASSERT_EQ(status.code, -1);
+        ASSERT_EQ(status.msg, "long_windows option doesn't match window in sql");
+    }
+
```
Comment (zhanghaohit) [inline src/sdk/sql_cluster_test.cc]: fix lint error

**Review by zhanghaohit — APPROVED** [review]
LGTM

**Review by dl239 — APPROVED** [review]

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**